hoverxref/extension.py

```diff
@@ -335,8 +321,13 @@ def deprecated_configs_warning(app, exception):
 def setup(app):
     """Setup ``hoverxref`` Sphinx extension."""
 
-    # ``override`` was introduced in 1.8
-    app.require_sphinx('1.8')
+    # The event listener structure used requires Sphinx 3.0 or newer
+    # Note that the pyproject.toml required 5.0 - different here because
+    # the restriction there is about ensuring users get dependencies that
+    # work. This restriction is the lowest version that this extension
+    # should work on if a user is willing to manually fiddle with dependency
+    # versions themselves.
+    app.require_sphinx('3.0')
```

Comment:
I wasn't sure whether to go with what I have here, or just restrict to 5 - happy to change if you prefer to just restrict to 5 (or if we want me to undo other changes to keep theoretical compatibility with much older versions).

Also, I've said '3.0` here, but `test_intersphinx_default_configs` would fail on Sphinx 3 with my refactor, as it requires 4.0 or newer now I've removed the branch. I think that makes more sense - users fiddling with older versions probably don't care about running the tests here, and when we're running the tests, we'll be obeying the version constraints in `pyproject.toml` in my mind. I can change though if you disagree.

Reply:
Thanks for the explanation. We can restrict to 5.0 here to match the requirements. Yeah.

Reply:
Done! (I just amended the previous commit to simplify the git history).